From 9414eac6365a7773fc8b544b3ea8f34f49b095ac Mon Sep 17 00:00:00 2001
From: Rolando Espinoza <rndmax84@gmail.com>
Date: Sun, 5 Mar 2017 14:11:15 -0300
Subject: [PATCH] Disable boost quadmath

Pystan passes its own flags to compile its models. If we don't disable
float128 explicitly then boost attempts to load quadmath.h header file.
---
 pystan/model.py | 1 +
 setup.py        | 1 +
 2 files changed, 2 insertions(+)

diff --git a/pystan/model.py b/pystan/model.py
index 790920f..5f9661d 100644
--- a/pystan/model.py
+++ b/pystan/model.py
@@ -271,6 +271,7 @@ class StanModel:
             ('BOOST_RESULT_OF_USE_TR1', None),
             ('BOOST_NO_DECLTYPE', None),
             ('BOOST_DISABLE_ASSERTS', None),
+            ('BOOST_MATH_DISABLE_FLOAT128', None),
         ]
         # compile stan models with optimization (-O2)
         # (stanc is compiled without optimization (-O0) currently, see #33)
diff --git a/setup.py b/setup.py
index d5af24b..88371d1 100644
--- a/setup.py
+++ b/setup.py
@@ -107,6 +107,7 @@ stan_macros = [
     ('BOOST_RESULT_OF_USE_TR1', None),
     ('BOOST_NO_DECLTYPE', None),
     ('BOOST_DISABLE_ASSERTS', None),
+    ('BOOST_MATH_DISABLE_FLOAT128', None),
     ('FUSION_MAX_VECTOR_SIZE', 12),  # for parser, stan-dev/pystan#222
 ]
 extra_compile_args = [
-- 
2.6.2

